components/match2/wikis/clashroyale/matchgroup_input_custom.lua

```diff
@@ -0,0 +1,747 @@
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local Flags = require('Module:Flags')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local CardNames = mw.loadData('Module:CardNames')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+local Streams = Lua.import('Module:Links/Stream', {requireDevIfEnabled = true})
+
+local Opponent = require('Module:OpponentLibraries').Opponent
+
+local UNKNOWN_REASON_LOSS_STATUS = 'L'
+local DEFAULT_WIN_STATUS = 'W'
+local DEFAULT_WIN_RESULTTYPE = 'default'
+local NO_SCORE = -1
+local SCORE_STATUS = 'S'
+local ALLOWED_STATUSES = {DEFAULT_WIN_STATUS, 'FF', 'DQ', UNKNOWN_REASON_LOSS_STATUS}
+local MAX_NUM_OPPONENTS = 2
+local DEFAULT_BEST_OF = 99
+local EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
+local NOW = os.time(os.date('!*t'))
+local ROYALE_API_PREFIX = 'https://royaleapi.com/'
+local MAX_NUM_PLAYERS_PER_MAP = 2
+local TBD = 'tbd'
+local TBA = 'tba'
+local MAX_NUM_MAPS = 30
+
+local CustomMatchGroupInput = {}
+
+CustomMatchGroupInput.walkoverProcessing = {}
+local walkoverProcessing = CustomMatchGroupInput.walkoverProcessing
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(match)
+	Table.mergeInto(
+		match,
+		CustomMatchGroupInput._readDate(match)
+	)
+	match = CustomMatchGroupInput._getExtraData(match)
+	match = CustomMatchGroupInput._getTournamentVars(match)
+	match = CustomMatchGroupInput._adjustData(match)
+	match = CustomMatchGroupInput._getVodStuff(match)
+	match = CustomMatchGroupInput._getLinks(match)
+
+	return match
+end
+
+function CustomMatchGroupInput._readDate(matchArgs)
+	if matchArgs.date then
+		return MatchGroupInput.readDate(matchArgs.date)
+	else
+		return {
+			date = EPOCH_TIME_EXTENDED,
+			dateexact = false,
+			timestamp = DateExt.epochZero,
+		}
+	end
+end
+
+function CustomMatchGroupInput._getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'solo'))
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function CustomMatchGroupInput._getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod)
+
+	return match
+end
+
+function CustomMatchGroupInput._getLinks(match)
+	match.links = {
+		royaleapi = match.royaleapi and (ROYALE_API_PREFIX .. match.royaleapi) or nil,
+	}
+	return match
+end
+
+function CustomMatchGroupInput._getExtraData(match)
+	match.extradata = {
+		mvp = MatchGroupInput.readMvp(match),
+		mvpteam = match.mvpteam or match.winner,
```

Comment:
needed?

Reply:
MVP stuff is the entire reason this PR exists. Request from the wiki to add MVPs

Reply:
i mean explicitly mvptean

Reply:
Oh, probably not. Ill take a deeper look tomorrow